Yank/experiment.py

```diff
@@ -1317,129 +1615,174 @@ def _validate_systems(self, systems_description):
+        solvent_dsl:
+            required: no
+            type: string
+            dependencies: [phase1_path, phase2_path]
+
+        # Phase paths
+        phase1_path:
+            required: no
+            type: list
+            schema:
+                type: string
+                validator: file_exists
+            dependencies: phase2_path
+            coerce: sort_alphabetically_by_extension
+        phase2_path:
+            required: no
+            type: list
+            schema:
+                type: string
+                validator: file_exists
+            validator: is_system_files_matching_phase1
+            coerce: sort_alphabetically_by_extension
+        solvent:
+            required: no
+            type: string
+            excludes: [solvent1, solvent2]
+            allowed: SOLVENT_IDS_POPULATED_AT_RUNTIME
+            validator: PIPELINE_SOLVENT_DETERMINED_AT_RUNTIME_WITH_RECEPTOR
+            oneof:
+                - dependencies: [phase1_path, phase2_path]
+                - dependencies: [receptor, ligand]
+        solvent1:
+            required: no
+            type: string
+            excludes: solvent
+            allowed: SOLVENT_IDS_POPULATED_AT_RUNTIME
+            validator: PIPELINE_SOLVENT_DETERMINED_AT_RUNTIME_WITH_SOLUTE
+            oneof:
+                - dependencies: [phase1_path, phase2_path, solvent2]
+                - dependencies: [solute, solvent2]
+        solvent2:
+            required: no
+            type: string
+            excludes: solvent
+            allowed: SOLVENT_IDS_POPULATED_AT_RUNTIME
+            validator: PIPELINE_SOLVENT_DETERMINED_AT_RUNTIME_WITH_SOLUTE
+            oneof:
+                - dependencies: [phase1_path, phase2_path, solvent2]
+                - dependencies: [solute, solvent2]
+        gromacs_include_dir:
+            required: no
+            type: string
+            dependencies: [phase1_path, phase2_path]
+            validator: directory_exists
+
+        # Non-Prebuilt setup
+        receptor:
+            required: no
+            type: string
+            dependencies: [ligand, solvent]
+            allowed: MOLECULE_IDS_POPULATED_AT_RUNTIME
+            excludes: [solute, phase1_path, phase2_path]
+            validator: REGION_CLASH_DETERMINED_AT_RUNTIME_WITH_LIGAND
+        ligand:
+            required: no
+            type: string
+            dependencies: [ligand, solvent]
+            allowed: MOLECULE_IDS_POPULATED_AT_RUNTIME
+            excludes: [solute, phase1_path, phase2_path]
+        pack:
+            # Technically requires receptor, but with default interjects itself even if receptor is not present
+            required: no
+            type: boolean
+            default: no
+        solute:
+            required: no
+            type: string
+            allowed: MOLECULE_IDS_POPULATED_AT_RUNTIME
+            dependencies: [solvent1, solvent2]
+            validator: REGION_CLASH_DETERMINED_AT_RUNTIME
```

Comment:
I dont actually think it can, but if we start allowing regions to be specified in the system/restraint this check will need to be here, so I left it in for when we eventually get to that point. This does not add too much overhead since it just goes through the regions in the solute and checks that those regions are not in an empty dictionary. 

so roughly the same complexity `O(2n)`: `[assert key in dict for key in dict.keys()]`

Reply:
Ah, I see. Sounds good!